pkg/controller/smart-clone-controller.go

```diff
@@ -335,16 +343,30 @@ func (c *SmartCloneController) updateSmartCloneStatusPhase(phase cdiv1.DataVolum
 		event.message = fmt.Sprintf(MessageCloneSucceeded, dataVolumeCopy.Spec.Source.PVC.Namespace, dataVolumeCopy.Spec.Source.PVC.Name, newPVC.Namespace, newPVC.Name)
 	}
 
-	return c.emitEvent(dataVolume, dataVolumeCopy, &event)
+	return c.emitEvent(dataVolume, dataVolumeCopy, &event, newPVC)
 }
 
-func (c *SmartCloneController) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, event *DataVolumeEvent) error {
+func (c *SmartCloneController) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, event *DataVolumeEvent, newPVC *corev1.PersistentVolumeClaim) error {
 	// Only update the object if something actually changed in the status.
 	if !reflect.DeepEqual(dataVolume.Status, dataVolumeCopy.Status) {
 		_, err := c.cdiClientSet.CdiV1alpha1().DataVolumes(dataVolume.Namespace).Update(dataVolumeCopy)
-		// Emit the event only when the status change happens, not every time
-		if event.eventType != "" {
-			c.recorder.Event(dataVolume, event.eventType, event.reason, event.message)
+		if err != nil {
```

Comment:
What was going on here?  Why not okay to just return the error?

Reply:
So what happens if we return the error, is that the data volume phase is not updated. So that will requeue the data volume. Both the data volume and smart controller will grab it, and the exact same thing will happen, right before we update in the smart controller, the data volume controller will finish syncing and we would fail again, and thus requeue again, etc, etc. So instead of always returning the error, I check for a conflict, do a recursive refresh the data volume, then update it while the data volume controller is done with this data volume and thus is not updating it.

Reply:
I understand how there can be contention between the two controllers but I don't see how a failure here causes the datavolume to be "requeued" in the datavolume controller.  If the update succeeds, then yes.  But not on failure. 
 Controllers should have independent queues.  And failures don't cause items to be "requeued" rather on success keys are removed from the queue.

Reply:
So it turns out to not be re-queuing on failure. If I force a re-queue it works.